sound/soc/sof/ipc4.c

```diff
@@ -637,6 +637,9 @@ static void sof_ipc4_rx_msg(struct snd_sof_dev *sdev)
 	case SOF_IPC4_NOTIFY_LOG_BUFFER_STATUS:
 		sof_ipc4_mtrace_update_pos(sdev, SOF_IPC4_LOG_CORE_GET(ipc4_msg->primary));
 		break;
+	case SOF_IPC4_NOTIFY_EXCEPTION_CAUGHT:
+		snd_sof_handle_fw_exception(sdev, "FW Panic");
```

Comment:
This is not a correct function to print the panic. The IPC4 exception is a firmware panic which is not recoverable.
you should really call
```c
snd_sof_dsp_panic(sdev, 0, true);
```
here!
If you happen to have an IPC timeout first, followed by an exception then only the IPC timeout will cause a DSP dump and the second call to `snd_sof_handle_fw_exception()` will skip the dump, see `sdev->dbg_dump_printed` flag in code.

Reply:
sure, I will updated it